### ISSUE TITLE
File names containing : are broken in 4.5.x

### ISSUE THREAD

**Issue by shane-axiom** [Unidata/thredds#110]
https://github.com/Unidata/thredds/blob/4.5.5/cdm/src/main/java/ucar/nc2/util/Misc.java#L300

Misc.getProtocols(String url) assumes that any `:` in a file name is a protocol delimiter. *nix OSes allow `:` in file names, so this needs to be fixed to eliminate false positives.

Currently this leads to some confusing exceptions where a `.dds` suffix is added to the filename:

```
Exception in thread "main" ucar.httpservices.HTTPException: Malformed URL: /blah/blah/some_file_2014-04-13_16:00:00.nc.dds
        at ucar.httpservices.HTTPMethod.<init>(HTTPMethod.java:198)
        at ucar.httpservices.HTTPMethod.<init>(HTTPMethod.java:187)
        at ucar.httpservices.HTTPFactory.Get(HTTPFactory.java:124)
        at ucar.nc2.dataset.NetcdfDataset.checkIfDods(NetcdfDataset.java:857)
        at ucar.nc2.dataset.NetcdfDataset.disambiguateHttp(NetcdfDataset.java:817)
        at ucar.nc2.dataset.NetcdfDataset.openOrAcquireFile(NetcdfDataset.java:707)
        at ucar.nc2.dataset.NetcdfDataset.acquireFile(NetcdfDataset.java:638)
        at ucar.nc2.ncml.Aggregation$Dataset.acquireFile(Aggregation.java:683)
        at ucar.nc2.ncml.AggregationExisting.buildNetcdfDataset(AggregationExisting.java:74)
        at ucar.nc2.ncml.Aggregation.finish(Aggregation.java:429)
        at ucar.nc2.ncml.NcMLReader.readNetcdf(NcMLReader.java:502)
        at ucar.nc2.ncml.NcMLReader._readNcML(NcMLReader.java:457)
        at ucar.nc2.ncml.NcMLReader.readNcML(NcMLReader.java:257)
        at ucar.nc2.ncml.NcMLReader.readNcML(NcMLReader.java:207)
        at ucar.nc2.dataset.NetcdfDataset.acquireNcml(NetcdfDataset.java:1085)
        at ucar.nc2.dataset.NetcdfDataset.openOrAcquireFile(NetcdfDataset.java:724)
        at ucar.nc2.dataset.NetcdfDataset.openDataset(NetcdfDataset.java:428)
        at ucar.nc2.dataset.NetcdfDataset.openDataset(NetcdfDataset.java:411)
        at ucar.nc2.dataset.NetcdfDataset.openDataset(NetcdfDataset.java:396)
        at ucar.nc2.dataset.NetcdfDataset.openDataset(NetcdfDataset.java:383)
```

[Ticket AYO-259890](http://www.unidata.ucar.edu/support/help/MailArchives/netcdf/msg13030.html) may also be related.

**Comment by DennisHeimbigner** [comment]
There is an inherent ambiguity in urls versus file paths. Technically, for most OSs, a url could be interpreted as a file path. So, anything that looks like a url is assumed to be one.
Note also that thredds complicates things by allowing a chain of protocols in its urls: e.g. proto1:proto2:...
However, your point is well taken and I will add code to look at colons more closely.
Do you have a specific example that you used to show the problem so I can incorporate
it into a test case?

Re: Ticket AYO-259890
A fix was added some time ago to check for windows drive letters. This problem is related.

**Comment by DennisHeimbigner** [comment]
Shane-
This fix is in 4.6.0 and backported to 4.5.6.  If you have a chance, please see if it fixes your problem.

**Comment by shane-axiom** [comment]
Confirmed fixed in 4.6.0. Thanks!
